### PR TITLE
Drop Zabbix 2/3 support and deprecated parameters; only support zabbix 4.0, 5.0, 5.2

### PR DESCRIPTION
**Pull request by bastelfreak**
Only Zabbix 4 and newer are supported upstream. This PR drops support
for older releases. See https://www.zabbix.com/de/life_cycle_and_release_policy
for current versions that upstream supports.

### CODE REVIEW

File: spec/classes/repo_spec.rb [hunk 1]
```diff
@@ -169,115 +100,74 @@
...
-                manage_repo: true
-              }
-            end
-
-            it { is_expected.to contain_yumrepo('zabbix').with_baseurl('https://repo.zabbix.com/zabbix/2.2/rhel/6/$basearch/') }
-            it { is_expected.to contain_yumrepo('zabbix').with_gpgkey('https://repo.zabbix.com/RPM-GPG-KEY-ZABBIX') }
-            it { is_expected.to contain_yumrepo('zabbix-nonsupported').with_baseurl('https://repo.zabbix.com/non-supported/rhel/6/$basearch/') }
-            it { is_expected.to contain_yumrepo('zabbix-nonsupported').with_gpgkey('https://repo.zabbix.com/RPM-GPG-KEY-ZABBIX') }
-          end
+        when '8'
```
Comment (root-expert) [inline spec/classes/repo_spec.rb:103]: We don't support RedHat 8

File: spec/classes/repo_spec.rb [hunk 2]
```diff
@@ -169,115 +100,74 @@
...
-            it { is_expected.to contain_yumrepo('zabbix-nonsupported').with_baseurl('https://repo.zabbix.com/non-supported/rhel/6/$basearch/') }
-            it { is_expected.to contain_yumrepo('zabbix-nonsupported').with_gpgkey('https://repo.zabbix.com/RPM-GPG-KEY-ZABBIX') }
+            it { is_expected.to contain_yumrepo('zabbix').with_baseurl('https://repo.zabbix.com/zabbix/5.0/rhel/8/$basearch/') }
+            it { is_expected.to contain_yumrepo('zabbix').with_gpgkey('https://repo.zabbix.com/RPM-GPG-KEY-ZABBIX-A14FE591') }
+            it { is_expected.to contain_yumrepo('zabbix-nonsupported').with_baseurl('https://repo.zabbix.com/non-supported/rhel/8/$basearch/') }
+            it { is_expected.to contain_yumrepo('zabbix-nonsupported').with_gpgkey('https://repo.zabbix.com/RPM-GPG-KEY-ZABBIX-79EA5ED4') }
           end
 
-          context 'on RedHat 6 and Zabbix 3.2' do
+          context 'on RedHat 8 and Zabbix 4.0' do
```
Comment (root-expert) [inline spec/classes/repo_spec.rb:133]: Same here

File: spec/classes/repo_spec.rb (hunk 1 above)
Comment (bastelfreak) [reply spec/classes/repo_spec.rb]: cleaned it up

File: spec/classes/repo_spec.rb (hunk 2 above)
Comment (bastelfreak) [reply spec/classes/repo_spec.rb]: cleaned it up

**Review by root-expert — APPROVED** [review]

**Comment by bastelfreak** [comment]
This PR also contains https://github.com/voxpupuli/puppet-zabbix/pull/751

File: README.md [hunk 3]
```diff
@@ -504,47 +504,9 @@ There are some zabbix specific parameters, please check them by opening the mani
...
-  * OracleLinux 5.x, 6.x
-  * Scientific Linux 5.x, 6.x
-  * Ubuntu 12.04
-  * Debian 6, 7
-  * xenserver 6
+
+This module supports Zabbix 4.0 to 5.2. The upstream supported versions are documented [here](https://www.zabbix.com/de/life_cycle_and_release_policy)
+Please have a look into the metadata.json for all supported operating systems.
 
 This module is supported on both the community as the Enterprise version of Puppet.
```
Comment (ekohl) [inline README.md:511]: Unrelated, but 
```suggestion
This module is supported on both the community and the Enterprise version of Puppet.
```

File: manifests/agent.pp [hunk 4]
```diff
@@ -321,20 +321,12 @@
...
   if $zabbix_version != $zabbix::params::zabbix_version and $service_type == $zabbix::params::service_type {
-    $real_service_type = versioncmp($zabbix_version, '3.0') ? {
-      1  => 'simple',
-      0  => 'simple',
-      -1 => 'forking',
-    }
+    $real_service_type = 'simple'
   } else {
     $real_service_type = $service_type
   }
```
Comment (ekohl) [inline manifests/agent.pp:332]: I don't understand this logic. If there is no differentiation on version anymore, can't you just trust `$service_type`? Same above with `additional_service_params`.

File: manifests/server.pp [hunk 5]
```diff
@@ -405,11 +382,7 @@
   # but only if the variable isn't provided.
 
   if $zabbix_version != $zabbix::params::zabbix_version and $additional_service_params == $zabbix::params::additional_service_params {
```
Comment (ekohl) [inline manifests/server.pp:384]: Same as earlier: can't you just trust the param now?

File: manifests/web.pp [hunk 6]
```diff
@@ -275,24 +275,9 @@
...
-      '3.2' : {
-        $zabbixapi_version = '3.2.1'
-      }
-      '3.4' : {
-        $zabbixapi_version = '4.0.0'
-      }
       '4.0': {
         $zabbixapi_version = '4.2.0'
       }
-      '4.4': {
```
Comment (ekohl) [inline manifests/web.pp:293]: This suggests 4.4 is no longer supported but 4.0 is. Is that correct?

File: manifests/agent.pp (hunk 4 above)
Comment (bastelfreak) [reply manifests/agent.pp]: ah good  catch. I cleaned that up

File: manifests/web.pp (hunk 6 above)
Comment (bastelfreak) [reply manifests/web.pp:293]: yep. 4.0 is the supported LTS. additionally supported are 5.0 and 5.2

**Review by ekohl — COMMENTED** [review]
Haven't done a very thorough review but I don't see more issues. The only thing is that I suspect that if previously the user set version to 4.4, that no longer works for web. The PR title doesn't make that clear which can surprise users.

**Comment by root-expert** [comment]
Maybe we should update the README with all the supported versions?

**Comment by Fabian1976** [comment]
Is it smart to restrict versions like that?
This module isn't know for its quick releases and patches (last release was more then 1 year ago). Zabbix release 5.4 is already out and will probably work fine with this module, except for the strict version lock which prevents it from running. I expect Zabbix to drop 5.2 support pretty soon (as they always do when a new, non LTS, version is released).

I hope the release cycle for this module will be sped up...

**Pull request merged**